src/ast/SemanticChecksVisitor.java

```diff
@@ -35,32 +42,294 @@ public void visit(Program program) {
 
     @Override
     public void visit(ClassDecl classDecl) {
+        if(classDecl.superName() != null){
+            String superName = classDecl.superName();
+            //semantic check - no variable with name already defined in ancestor (req 4)
+            Set<String> parentFieldNames = new HashSet<>();
+            for(STSymbol symbol : instanceTemplates.get(superName)) parentFieldNames.add(symbol.name());
```

Comment:
I don't think so because we need the names of the variables and that comes from the List of the parent.
The keySet of the man will be the names of all the classes - we don't need that here